web/src/logfilters.ts

```diff
@@ -23,9 +23,37 @@ export enum FilterSource {
   runtime = "runtime",
 }
 
+export enum TermState {
+  Empty = "empty",
+  Parsed = "parsed",
+  Error = "error",
+}
+
+type EmptyTerm = { state: TermState.Empty }
+
+type ParsedTerm = { state: TermState.Parsed; regex: RegExp }
+
+type ErrorTerm = { state: TermState.Error; error: string }
+
+type FilterTerm = {
+  input: string // Unmodified string input
+} & (EmptyTerm | ParsedTerm | ErrorTerm)
+
 export type FilterSet = {
   level: FilterLevel
   source: FilterSource
+  term: FilterTerm
+}
+
+export const EMPTY_TERM = ""
+export const EMPTY_FILTER_TERM: FilterTerm = {
+  input: EMPTY_TERM,
+  state: TermState.Empty,
+}
+
+export function parseFilterTerm(term: string): RegExp {
+  // Filter terms are case-insensitive and can match multiple instances
+  return new RegExp(term, "gi")
```

Comment:
(Commenting to document a discussion @landism and I had yesterday)

My plan was to transform a user's input string into a regex. That input can either be treated as a string literal or as a regex and passed to the `RegExp` constructor, but downstream consumers don't need to worry about what kind of filter term a user enters; they'll have the same interface to filter text by.

For example:
*  the user input `fail(ed)?` would be `new RegExp("fail\\(ed\\)\\?")` and match only on strings that include that exact string
*  while the user input `/fail(ed)?/` would be `new RegExp("fail(ed)?")` and match strings that include `fail` and `failed`

I had incorrectly assumed based on incomplete testing on my part that I could pass this input directly to the `RegExp` constructor, and it would treat the input as a string literal unless surrounded by `/`. That's not the case. 😅 Any regex-specific characters need to be escaped. I'm going to implement this and add tests around these cases.